pkg/operator/ceph/pool/controller.go

```diff
@@ -173,6 +182,14 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 			return reconcile.Result{}, errors.Wrapf(err, "failed to delete pool %q. ", cephBlockPool.Name)
 		}
 
+		// disable RBD stats collection if cephBlockPool was deleted
+		cephBlockPool.Spec.EnableRBDStats = false
+		logger.Debug("configuring RBD per-image IO statistics collection")
```

Comment:
Let's add the logging inside the `configureRBDStats()` method so it doesn't need to be duplicated everywhere it is called.